packages/react/src/hooks/actions/useNavigateAction.ts

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { Hub } from '@aws-amplify/core';
+
+import {
+  ACTIONS_CHANNEL,
+  ACTION_NAVIGATE_FINISHED,
+  ACTION_AUTH_SIGNOUT_STARTED,
+  ACTION_NAVIGATE_STARTED,
+} from './constants';
+
+type NavigateType = 'url' | 'anchor' | 'reload';
+
+type NavigateRun = () => void;
+
+export interface UseNavigateActionProps {
+  type: NavigateType;
+
+  url?: string;
+
+  anchor?: string;
+
+  target?: React.HTMLAttributeAnchorTarget;
+}
+
+export const defaultTarget = '_self';
+
+export const useNavigateAction = (props: UseNavigateActionProps) => {
+  const { type, url, anchor, target } = props;
+  let run: NavigateRun;
```

Comment:
Should this be wrapped in useCallback since hooks are run on every render and we don't want to recreate the `run` function this unless the props change?

Reply:
Thats a good point. Will update